concordium-consensus/tests/scheduler/SchedulerTests/TestUtils.hs

```diff
@@ -85,16 +85,16 @@ defaultParams = TestParameters
 -- | A test case for executing a list of transactions, specifying 'ResultSpec's for the result
 -- of each transaction's execution. The transactions are run with 'Sch.filterTransactions'
 -- in sequenced and not grouped, with the given parameters.
-data TestCase = TestCase
+data TestCase pv = TestCase
```

Comment:
Perhaps a test case should existentially quantify the protocol version, subject to `IsProtocolVersion pv`. That way, `mkSpecs` can process lists of test cases of multiple versions.